### PR TITLE
Fix tracking learning

### LINKED ISSUE #2052: Tracking with learning: some leftover networkx 1.x code
we'll have to ensure no networkx 1.x syntax (e.g. `graph.node[key]`) is used anymore. Should not be closed without a test that runs when solvers are installed.

### PR DESCRIPTION
**Pull request by k-dominik**
Addresses a few issues related to learning tracking weights in Tracking.

Summary:

* there was some incomplete refactor from `networkx 1.x` to `2.x`
* added tests

Todo:
* [x] refactor `insertAnnotationsToHypothesesGraph`
* [x] rebuild hytra (needed, see https://github.com/ilastik/hytra/pull/8)

fixes #2052, #2056, #2057, #2005

### CODE REVIEW

File: ilastik/applets/tracking/structured/opStructuredTracking.py [hunk 1]
```diff
@@ -565,36 +572,49 @@ def insertAnnotationsToHypothesesGraph(self, traxelgraph, annotations,misdetecti
...
+                for destination_object_id in labels[t+1].keys():
+                    # skip if misdetection inside
+                    if (misdetectionLabel in labels[t+1][destination_object_id]):
+                        continue
+
+                    intersectSet = labels[t][source_object_id].intersection(labels[t+1][destination_object_id])
+                    lenIntersectSet = len(intersectSet)
+                    if lenIntersectSet > 0:
+                        if ((t, source_object_id), (t+1, destination_object_id)) not in traxelgraph._graph.edges.keys():
+                            raise AnnotationHypothesisgraphMismatchException(
```
Comment (emilmelnikov) [inline ilastik/applets/tracking/structured/opStructuredTracking.py]: Consider moving the message string into `__init__`:

```python
class SpamException:
    def __init__(self, src_obj_id, track_ids):
        msg = f"... {src_obj_id} ... {track_ids} ..."
        super().__init__(msg)
````

File: ilastik/applets/tracking/structured/opStructuredTracking.py [hunk 2]
```diff
@@ -531,19 +535,22 @@ def _type(self, time, track):
             elif type != None:
                 return [type]
 
-    def getLabel(self, time, track, labels):
+    @staticmethod
+    def getLabel(time, track, labels):
         for label in list(labels[time].keys()):
```
Comment (emilmelnikov) [inline ilastik/applets/tracking/structured/opStructuredTracking.py]: If we are here anyway, maybe refactor this and `getLabelT` along the way?

```python
return next(label for label, v in labels[time].items() if len(v) == 1 and track in v, False)
```

File: tests/test_applets/structuredLearningTracking/testOpStructuredTracking.py [hunk 3]
```diff
@@ -0,0 +1,204 @@
...
+            return False
+        if self.Timestep == other.Timestep:
+            if self.Id == other.Id:
+                return True
+
+        return False
+
+
+@pytest.fixture
+def expected():
```
Comment (emilmelnikov) [inline tests/test_applets/structuredLearningTracking/testOpStructuredTracking.py]: Do we need to keep "nodes" and "edges" structures together? We could split them into 2 different fixtures.

Comment (k-dominik) [reply tests/test_applets/structuredLearningTracking/testOpStructuredTracking.py]: you're absolutely right. I kept them together as I was just understanding the tracking code, and there annotations are kept together in a dict. But here there is no benefit, other than mirroring the data structures used in the code.

**Comment by k-dominik** [comment]
thx @emilmelnikov for the suggestions! I moved the exception message out but will keep the data structure mirroring the code.

File: tests/test_applets/structuredLearningTracking/testOpStructuredTracking.py (hunk 3 above)
Comment (emilmelnikov) [reply tests/test_applets/structuredLearningTracking/testOpStructuredTracking.py]: Oh, I missed that, sorry. If it mirrors a structure in SUT, then definitely keep it as-is!

**Pull request merged**